src/probeinterface/probe.py

```diff
@@ -960,6 +961,181 @@ def from_numpy(arr: np.ndarray) -> "Probe":
 
         return probe
 
+    def add_probe_to_zarr_group(self, group: "zarr.Group") -> None:
+        """
+        Serialize the probe's data and structure to a specified Zarr group.
+
+        This method is used to save the probe's attributes, annotations, and other
+        related data into a Zarr group, facilitating integration into larger Zarr
+        structures.
+
+        Parameters
+        ----------
+        group : zarr.Group
+            The target Zarr group where the probe's data will be stored.
+        """
+
+        # Top-level attributes used to initialize a new Probe instance
+        group.attrs["ndim"] = self.ndim
+        group.attrs["si_units"] = self.si_units
+
+        # Special attributes
+        group.attrs["name"] = self.name
+        group.attrs["manufacturer"] = self.manufacturer
+        group.attrs["model_name"] = self.model_name
+        group.attrs["serial_number"] = self.serial_number
+
+        # Annotations as a group
+        annotations_group = group.create_group("annotations")
+        for key, value in self.annotations.items():
+            annotations_group.attrs[key] = value
+
+        # Contact annotations as a group
+        contact_annotations_group = group.create_group("contact_annotations")
+        for key, value in self.contact_annotations.items():
+            contact_annotations_group.create_dataset(key, data=value, chunks=False)
+
+        # Save the following fields of the probe as top-level datasets
+        dataset_attrs = [
+            "_contact_positions",
+            "_contact_plane_axes",
+            "_contact_shapes",
+            "device_channel_indices",
+            "_contact_ids",
+            "_shank_ids",
+            "probe_planar_contour",
+        ]
+        for attr in dataset_attrs:
+            value = getattr(self, attr)
+            if value is not None:
+                group.create_dataset(attr, data=value, chunks=False)
+
+        # Handling contact_shape_params
+        if self._contact_shape_params is not None:
+            shape_params_json_np = np.array([json.dumps(d) for d in self._contact_shape_params], dtype=object)
+            group.create_dataset("contact_shape_params", data=shape_params_json_np, dtype=str)
```

Comment:
Why not
```suggestion
            shape_params = np.array([d for d in self._contact_shape_params], dtype=object)
            group.create_dataset("contact_shape_params", data=shape_params, object_codec=numcodecs.JSON())
```
Then we live numcodec ùake the json stuff. Maybe not because numcodec aadd another dep.
I don't know.

Reply:
We could also save contact_shape params as vector like in the dataframe representation no ?

Reply:
I think that numcodecs comes with zarr, is that correct, @alejoe91?


I think this is a good idea. I will check the dataframe representation.